client/resource_group/controller/controller.go

```diff
@@ -491,31 +506,14 @@ func (gc *groupCostController) initRunState() {
 	}
 }
 
-func (gc *groupCostController) updateRunState(ctx context.Context) {
+func (gc *groupCostController) updateRunState() {
 	newTime := time.Now()
-	deltaConsumption := &rmpb.Consumption{}
+	gc.mu.Lock()
 	for _, calc := range gc.calculators {
-		calc.Trickle(ctx, deltaConsumption)
+		calc.Trickle(gc.mu.consumption, gc.mainCfg.isSingleGroupByKeyspace)
```

Comment:
Currently, all `ResourceCalculator`s own the config inside, so there is no need to pass it again here.